src/Kafka/Consumer/Callbacks.hs

```diff
@@ -14,16 +14,17 @@ import Kafka.Callbacks        as X
 import Kafka.Consumer.Convert (fromNativeTopicPartitionList', fromNativeTopicPartitionList'')
 import Kafka.Consumer.Types   (KafkaConsumer (..), RebalanceEvent (..), TopicPartition (..))
 import Kafka.Internal.RdKafka
-import Kafka.Internal.Setup   (HasKafka (..), HasKafkaConf (..), Kafka (..), KafkaConf (..), getRdMsgQueue)
+import Kafka.Internal.Setup   (HasKafka (..), HasKafkaConf (..), Kafka (..), KafkaConf (..), getRdMsgQueue, Callback (..))
 import Kafka.Types            (KafkaError (..), PartitionId (..), TopicName (..))
 
 import qualified Data.Text as Text
 
 -- | Sets a callback that is called when rebalance is needed.
-rebalanceCallback :: (KafkaConsumer -> RebalanceEvent -> IO ()) -> KafkaConf -> IO ()
-rebalanceCallback callback kc@(KafkaConf conf _ _) = rdKafkaConfSetRebalanceCb conf realCb
+rebalanceCallback :: (KafkaConsumer -> RebalanceEvent -> IO ()) -> Callback
+rebalanceCallback callback =
+  Callback $ \kc@(KafkaConf con _ _) -> rdKafkaConfSetRebalanceCb con (realCb kc)
   where
-    realCb k err pl = do
+    realCb kc k err pl = do
```

Comment:
Any reason why you also renamed this variable?

Reply:
The issue here is that because the `KafkaConf` argument moves into the newtype, the variable is now bound in the lambda instead of the function definition:

```
   Callback $ \kc@(KafkaConf con _ _) -> rdKafkaConfSetRebalanceCb con (realCb kc)
```

This means `kc` is no longer available in the where-clause, hence I added it as an extra argument to `realCb`. The other obvious alternative would be to move `realCb` into a `let`, but I am not sure that is better.

Reply:
My apologies, I misread, I thought `k` was renamed to `kc` but it's another argument